tasks.go

```diff
@@ -0,0 +1,264 @@
+		return "", errors.New("не заполнен заголовок задачи")
+	}
+	now := time.Now()
+	nowString := now.Format("20060102")
+	date := task.Date
+	if len(date) == 0 {
+		date = now.Format("20060102")
+	}
+	_, err := time.Parse("20060102", date)
+	if err != nil {
+		log.Println(err)
+		return "", errors.New("некорректный формат даты")
+	}
+	if date < nowString && len(task.Repeat) > 0 {
+		date, err = nextDate(now, date, task.Repeat)
+		if err != nil {
+			return "", err
+		}
+	} else if date < nowString && len(task.Repeat) == 0 {
+		date = nowString
+	}
+
+	db, err := openDb()
+	if err != nil {
+		return "", err
+	}
+	defer db.Close()
+
+	res, err := db.Exec("INSERT INTO scheduler (date, title, comment, repeat) VALUES (:date, :title, :comment, :repeat)",
+		sql.Named("date", date),
+		sql.Named("title", task.Title),
+		sql.Named("comment", task.Comment),
+		sql.Named("repeat", task.Repeat))
+	if err != nil {
+		log.Println(err)
+		return "", errors.New("internal server error")
+	}
+	id, err := res.LastInsertId()
+	if err != nil {
+		log.Println(err)
+		return "nil", errors.New("internal server error")
+	}
+	return strconv.Itoa(int(id)), nil
+}
+
+func getTasks(search string) ([]task, error) {
+	db, err := openDb()
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+
+	query := "SELECT id, date, title, comment, repeat FROM scheduler"
+	var rows *sql.Rows
+	var tasks []task
+	if len(search) > 0 {
+		date, err := time.Parse("02.01.2006", search)
+		if err == nil {
+			query += " WHERE date = :search"
+			search = date.Format("20060102")
+		} else {
+			query += " WHERE title LIKE :search OR comment LIKE :search"
+			search = "%" + search + "%"
+		}
+		query += " ORDER BY date LIMIT 50"
+		rows, err = db.Query(query, sql.Named("search", search))
+		if err == sql.ErrNoRows {
+			return tasks, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		query += " ORDER BY date LIMIT 50"
+		rows, err = db.Query(query)
+		if err == sql.ErrNoRows {
+			return tasks, nil
+		}
+		if err != nil {
+			log.Println(err)
+			return nil, errors.New("internal server error")
+		}
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		task := task{}
+		err = rows.Scan(&task.Id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+		if err != nil {
+			log.Println(err)
+			return nil, errors.New("internal server error")
+		}
+		tasks = append(tasks, task)
+	}
+	return tasks, nil
+}
+
+func getTask(id string) (task, error) {
+	var task task
+	if len(id) == 0 {
+		return task, errors.New("не указан id")
+	}
+	db, err := openDb()
+	if err != nil {
+		return task, err
+	}
+	defer db.Close()
+	idInt, err := strconv.Atoi(id)
+	if err != nil {
+		log.Println(err)
+		return task, errors.New("не верный формат id")
+	}
+	query := "SELECT id, date, title, comment, repeat FROM scheduler WHERE id = :id"
+	row := db.QueryRow(query, sql.Named("id", idInt))
+	err = row.Scan(&task.Id, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+	if err == sql.ErrNoRows {
```

Comment:
Ошибки лучше сравнивать используя специальные функции
```suggestion
	if errors.Is(err, sql.ErrNoRows) {
```

Reply:
Исправил.